lib/index.js

```diff
@@ -179,6 +201,26 @@ FSTree.defaultIsEqual = function defaultIsEqual(entryA, entryB) {
   return equal;
 };
 
+FSTree.applyPatch = function(input, output, patch, _delegate) {
+  var delegate = assign({}, DEFAULT_DELEGATE, _delegate);
+  for (var i = 0; i < patch.length; i++) {
+    applyOperation(input, output, patch[i], delegate);
+  }
+};
+
+function applyOperation(input, output, operation, delegate) {
+  var method = operation[0];
+  var relativePath = operation[1];
+  var inputPath = path.join(input, relativePath);
+  var outputPath = path.join(output, relativePath);
+
+  if (delegate[method]) {
+    delegate[method](inputPath, outputPath, relativePath);
+  } else {
+    throw new Error('Unknown patch operation method: ' + method);
```

Comment:
This is always user error right?

We define the set of possible patch operations, and we provide a default implementation for all of them in `DEFAULT_DELEGATE`.  So the only way we get to `!delegate[method]` if if the user does something like give us a delegate with `{ create: false, change: null }` or similar.

We might be able to give a more precise error message here.

```js
throw new Error(`delegate.${method} is a ${typeof delegate[method]}, and not a function.  Check the 'delegate' argument to 'FSTree.prototype.applyPatch'.`);
```

thoughts?